From 7abcba27d325f5339b2734fd4bbd7ba4621adad4 Mon Sep 17 00:00:00 2001
From: S-trace <S-trace@list.ru>
Date: Sun, 17 Jan 2016 12:35:34 +0300
Subject: [PATCH 5/5] [DATAMEDIA] Set datamedia if /data/media/0/TWRP/.twrps
 config found

Don't lock out backups in datamedia with wiped /data/

Change-Id: I142c7916734b4e3b79f041184de4144ecced4fc1
---
 data.cpp | 7 ++++++-
 twrp.cpp | 3 +++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/data.cpp b/data.cpp
index 250b9a0..b7ba0cd 100644
--- a/data.cpp
+++ b/data.cpp
@@ -702,7 +702,12 @@ void DataManager::SetDefaultValues()
 		mConstValues.insert(make_pair(TW_HAS_DATA_MEDIA, "1"));
 		mConstValues.insert(make_pair("tw_has_internal", "1"));
 		datamedia = true;
-        } else {
+        } else if (access("/data/media/0/TWRP/.twrps", F_OK) == 0) {
+		printf("RECOVERY_SDCARD_ON_DATA := true (forced by /data/media/0/TWRP/.twrps\n");
+		mConstValues.insert(make_pair(TW_HAS_DATA_MEDIA, "1"));
+		mConstValues.insert(make_pair("tw_has_internal", "1"));
+		datamedia = true;
+	} else {
 		mValues.insert(make_pair(TW_HAS_DATA_MEDIA, make_pair("0", 0)));
 		mValues.insert(make_pair("tw_has_internal", make_pair("0", 0)));
         }
diff --git a/twrp.cpp b/twrp.cpp
index 253314a..21bb783 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -109,6 +109,9 @@ int main(int argc, char **argv) {
 	if (strncmp(usbmsc_present, "false", PROPERTY_VALUE_MAX) == 0 || strncmp(storage_configuration, STORAGES_CONFIGURATION_DATAMEDIA, PROPERTY_VALUE_MAX) == 0) {
 		printf("datamedia = true (forced by property ro.usbmsc.present=%s and persist.storages.configuration=%s\n", usbmsc_present, storage_configuration);
 		datamedia = true;
+	} else if (access("/data/media/0/TWRP/.twrps", F_OK) == 0) {
+		printf("datamedia = true (forced by /data/media/0/TWRP/.twrps\n");
+		datamedia = true;
 	}
 #endif
 
-- 
2.5.0

